### PR TITLE
Make socket keepAlive default to true

### PR DESCRIPTION
**Pull request by rozza**
JAVA-2531

### CODE REVIEW

File: driver-core/src/main/com/mongodb/connection/SocketSettings.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public static Builder builder() {
     public static class Builder {
         private long connectTimeoutMS = 10000;
         private long readTimeoutMS;
-        private boolean keepAlive;
+        private boolean keepAlive = true;
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/connection/SocketSettings.java]: According to spec, keepalive property (getters/setters) should be deprecated.

File: driver/src/main/com/mongodb/MongoClientOptions.java [hunk 2]
```diff
@@ -880,7 +880,7 @@ public String toString() {
         private int maxConnectionLifeTime;
         private int connectTimeout = 1000 * 10;
         private int socketTimeout = 0;
-        private boolean socketKeepAlive = false;
+        private boolean socketKeepAlive = true;
```
Comment (jyemin) [inline driver/src/main/com/mongodb/MongoClientOptions.java]: As above.  The getter and setter should be deprecated.

File: driver-core/src/main/com/mongodb/connection/SocketSettings.java (hunk 1 above)
Comment (rozza) [inline driver-core/src/main/com/mongodb/connection/SocketSettings.java]: Done

File: driver/src/main/com/mongodb/MongoClientOptions.java (hunk 2 above)
Comment (rozza) [inline driver/src/main/com/mongodb/MongoClientOptions.java]: Done

**Pull request closed without merging**